dashboard/lib/vocabulary_autocomplete.rb

```diff
@@ -0,0 +1,13 @@
+class VocabularyAutocomplete < AutocompleteHelper
+  def self.get_search_matches(query, limit, course_version_id)
+    limit = format_limit(limit)
+
+    rows = Vocabulary.limit(limit)
+    rows = rows.where(course_version_id: course_version_id)
+    return [] if query.length < MIN_WORD_LENGTH
+    query = format_query(query)
+    rows = rows.
+      where("MATCH(word,definition) AGAINST(? in BOOLEAN MODE)", query)
```

Comment:
I totally missed all the work that this line is doing when I reviewed the resources search PR: https://dev.mysql.com/doc/refman/8.0/en/fulltext-search.html Nice find on mysql's fulltext search! 

I am just wondering if you tried the "natural language" mode and had anything to share about why you picked boolean mode? no worries if not, but could be good to capture if so.

Reply:
I didn't -- for school search (what this logic was initially used for) natural language doesn't make much sense as it was just looking for a school or city name match. I just tried it out and it looks like the results might be better sorted but we lose searching by partial words (eg searching "algo" and "algorithm" popping up).

Reply:
Makes sense. Let's see how it goes, and we can always let the curriculum team try out the other way later if there are problems.

Reply:
Sounds good!